deploy/examples/import-external-cluster.sh

```diff
@@ -123,18 +123,24 @@ eof
 }
 
 function importClusterID() {
+  echo "Entering importClusterID"
   if [ -n "$RADOS_NAMESPACE" ]; then
+    echo "Creating rados namespace"
     createRadosNamespaceCR
-    timeout 20 sh -c "until [ $($KUBECTL -n "$NAMESPACE" get CephBlockPoolRadosNamespace/"$RADOS_NAMESPACE" -o jsonpath='{.status.phase}' | grep -c "Ready") -eq 1 ]; do echo "waiting for radosNamespace to get created" && sleep 1; done"
+    timeout 20 bash -c "while ! $KUBECTL -n \"$NAMESPACE\" get CephBlockPoolRadosNamespace/\"$RADOS_NAMESPACE\" -o jsonpath='{.status.phase}' | grep -q \"Ready\"; do echo \"waiting for radosNamespace to get created\"; sleep 1; done"
```

Comment:
I think I can see the issue that is being resolved here. I have some optimizations though, for keeping readability higher.

Kubernetes resource names can't have spaces in them, so there is no **need** to quote them. For legibility, I would suggest just getting rid of those quotes entirely. Since this is already within a double quoted string, I don't think shellcheck will complain.

For grep, the double quotes aren't necessary. Single quotes can be used instead. Ditto for the "waiting for..." message.

```suggestion
    timeout 20 bash -c "while ! $KUBECTL -n $NAMESPACE get CephBlockPoolRadosNamespace/$RADOS_NAMESPACE -o jsonpath='{.status.phase}' | grep -q 'Ready'; do echo 'waiting for radosNamespace to get created'; sleep 1; done"
```

Ditto this for the other changes as well.

Reply:
This is actually an essential change: as is, the script throws a syntax error. It is unusable. Another possibility is to change the outer double quotes to single ones:

```
-    timeout 20 sh -c "until [ $($KUBECTL -n "$NAMESPACE" get CephBlockPoolRadosNamespace/"$RADOS_NAMESPACE" -o jsonpath='{.status.phase}' | grep -c "Ready") -eq 1 ]; do echo "waiting for radosNamespace to get created" && sleep 1; done"
+    timeout 20 sh -c 'until [ $($KUBECTL -n "$NAMESPACE" get CephBlockPoolRadosNamespace/"$RADOS_NAMESPACE" -o jsonpath='{.status.phase}' | grep -c "Ready") -eq 1 ]; do echo "waiting for radosNamespace to get created" && sleep 1; done'
```

and

```
-    timeout 20 sh -c "until [ $($KUBECTL -n "$NAMESPACE" get CephFilesystemSubVolumeGroup/"$SUBVOLUME_GROUP" -o jsonpath='{.status.phase}' | grep -c "Ready") -eq 1 ]; do echo "waiting for radosNamespace to get created" && sleep 1; done"
+    timeout 20 sh -c 'until [ $($KUBECTL -n "$NAMESPACE" get CephFilesystemSubVolumeGroup/"$SUBVOLUME_GROUP" -o jsonpath='{.status.phase}' | grep -c "Ready") -eq 1 ]; do echo "waiting for radosNamespace to get created" && sleep 1; done'
```